src/Refactoring-UI-Tests/ReExtractToTemporaryDriverTest.class.st

```diff
@@ -0,0 +1,88 @@
+Class {
+	#name : 'ReExtractToTemporaryDriverTest',
+	#superclass : 'ReDriverTest',
+	#category : 'Refactoring-UI-Tests-Driver',
+	#package : 'Refactoring-UI-Tests',
+	#tag : 'Driver'
+}
+
+{ #category : 'initialization' }
+ReExtractToTemporaryDriverTest >> setUpMocksOn: driver [
+
+    | requestClass dialog |
+    dialog := MockObject new.
+    
+    dialog 
+        on: #variableName
+        respond: (RBArgumentName name: 'tempVar').
+        
+    dialog
+        on: #variableName
+        respond: (RBArgumentName name: #anotherVar).
+    
+    driver requestDialog: dialog.
```

Comment:
We craft these mocks based on how they are used in the driver. I'm not sure they are configured correctly. Basically what we do is create mock object. Go in the class and see exactly which messages are sent to this object. Each of them define here in `on:respond:`. Pay attention which message should return something meaningful, and which ones you don't care (if you don't care, return `self`).
For example `requestDialog:` receives `name:`, `label:` and `validateAnswer:`, right?

Main disadvantage of mocks is that they are tightly coupled to implementation and I'm not a big fan of them, but this is only way we found to test drivers.